lib/services/template-service.ts

```diff
@@ -83,97 +82,89 @@ export class TemplateService implements ITemplateService {
     }
 
     getTemplates(): Promise<Array<any>> {
-        // let tempDetails: Array<any> = [];
-        // const promises: Array<any> = [];
+        let templateDetails: Array<any> = [];
+        const promises: Array<any> = [];
 
         return new Promise((resolve, reject) => {
-            resolve(Backup.fallback);
-            // tmpCache.get("tempDetails", (error: any, value: any) => {
-            //     if (!error) {
-            //         if (value === undefined) {
-            //             this.getTemplatesNames()
-            //                 .then((templateNames: any) => {
-            //                     return templateNames;
-            //                 })
-            //                 .then((names: any) => {
-            //                     names.forEach((name: string) => {
-            //                         promises.push(
-            //                             this.getAppTemplateDetails(name)
-            //                                 .then((details) => {
-            //                                     tempDetails.push(details);
-            //                                 })
-            //                                 .catch((errorDetails) => {
-            //                                     reject(errorDetails);
-            //                                 })
-            //                         );
-            //                     });
-
-            //                     Promise.all(promises)
-            //                         .then(() => {
-            //                             tempDetails = this.sortTmpData(tempDetails);
-            //                             tmpCache.set("tempDetails", tempDetails, Config.cacheTime);
-            //                             resolve(tempDetails);
-
-            //                         })
-            //                         .catch((errorPromises: any) => {
-            //                             // TODO Implement error logger
-            //                             resolve(Backup.fallback);
-            //                         });
-            //                 })
-            //                 .catch((errorTemplates: any) => {
-            //                     console.error(errorTemplates);
-            //                 });
-            //         } else {
-            //             // Load data from cache
-            //             resolve(value);
-            //         }
-            //     }
-            // });
+            templateCache.get("templateDetails", (error: any, value: any) => {
```

Comment:
Ideas for refactoring (as a separate PR):

- reverse the "master" if clause from:
``` ts
if (!error) {
...
} else {
   resolve(...)
}
```
to
``` ts
if (error) {
   resolve(...)
   return;
}

...
```
In this way you are decreasing the indentation with one step for the bulk of the method.

Also, we should consider whether it is better to write single-line thennables with the short form:
``` ts
then((details) => templateDetails.push(details)
```
vs
``` ts
then((details) => {
  templateDetails.push(details);
}
```